eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java

```diff
@@ -2059,6 +2059,28 @@ default void appendString(Appendable appendable, String separator)
         return this.collectWith(function, parameter, target);
     }
 
+    /**
+     * This method will count the number of occurrences of each value calculated by applying the
+     * function to each element of the collection.
+     *
+     * @since 10.0.0
+     */
+    default <V> Bag<V> countByEach(Function<? super T, ? extends Iterable<V>> function)
+    {
+        return this.asLazy().<V>flatCollect(function).toBag();
+    }
+
+    /**
+     * This method will count the number of occurrences of each value calculated by applying the
+     * function to each element of the collection.
+     *
+     * @since 10.0.0
+     */
+    default <V, R extends MutableBagIterable<V>> R countByEach(Function<? super T, ? extends Iterable<V>> function, R target)
+    {
+        return this.<V, R>flatCollect(function, target);
```

Comment:
This overload `countByEach(function, target)`, just delegates to `flatCollect(function, target)`. I don't think we should add the method because it doesn't do any additional work and it's not a more intuitive alias than flatCollect, in my opinion. Thoughts?

Reply:
The target may add symmetry, I don't quite have a strong opinion in either direction.